setup.py

```diff
@@ -35,15 +37,84 @@ def is_conda_interpreter():
     """
     return 'conda' in sys.version or 'Continuum' in sys.version
 
+def is_macOS():
+    return 'darwin' in sys.platform
+
+def compile_link_paths_from_manifest(cmd):
+    """Parse include and library paths from OSX package managers.
+
+    Example executions::
+
+        # Homebrew
+        $ brew ls --verbose libomp
+        /opt/homebrew/Cellar/libomp/15.0.7/INSTALL_RECEIPT.json
+        /opt/homebrew/Cellar/libomp/15.0.7/.brew/libomp.rb
+        /opt/homebrew/Cellar/libomp/15.0.7/include/ompt.h
+        /opt/homebrew/Cellar/libomp/15.0.7/include/omp.h
+        /opt/homebrew/Cellar/libomp/15.0.7/include/omp-tools.h
+        /opt/homebrew/Cellar/libomp/15.0.7/lib/libomp.dylib
+        /opt/homebrew/Cellar/libomp/15.0.7/lib/libomp.a
+
+        # MacPorts
+        $ port contents libomp
+        Port libomp contains:
+          /opt/local/include/libomp/omp-tools.h
+          /opt/local/include/libomp/omp.h
+          /opt/local/include/libomp/ompt.h
+          /opt/local/lib/libomp/libgomp.dylib
+          /opt/local/lib/libomp/libiomp5.dylib
+          /opt/local/lib/libomp/libomp.dylib
+          /opt/local/share/doc/libomp/LICENSE.TXT
+          /opt/local/share/doc/libomp/README.txt
 
-# Get OpenMP setting from environment
-try:
-    use_omp = int(os.environ['USE_OMP'])
-except KeyError:
-    # OpenMP is not supported with default clang
-    # Conda provides its own compiler which does support openmp
-    use_omp = 'darwin' not in sys.platform or is_conda_interpreter()
+    """
+    from subprocess import run
+    query = run(cmd, shell=True, check=False, capture_output=True)
+    if query.returncode != 0:
+        return None, None
```

Comment:
I'm debating whether or not this should be an exception and the calling function should have a try/except. Maybe for this small utility it doesn't matter.

Reply:
I went with the return check on the principle that exceptions are abnormal and sometimes expensive paths of execution, and at least one of the two manifest providers is expected to fail. This may not be in line with current Python best practices.

Reply:
Yeah I usually avoid exceptions in cases like this too. I think the thing that made me suggest/bring up exceptions in the first place was that I wanted to add an error/print message here to say "running this command failed", but as you said this is expected for at least one of the cases. Let's leave it as is.